src/main/java/wromaciej/hvac_sim/thermo/devices/service/DefaultItemFactory.java

```diff
@@ -0,0 +1,101 @@
+        this.allElements = allElements;
+        this.unitSystem = unitSystem;
+        this.idGenerator = idGenerator;
+        this.allSolvers = allSolvers;
+    }
+
+    public HeatStream createDefaultHeatStream(){
+        HeatStream heatStream;
+        int uniqueId = idGenerator.getUniqueId();
+        Parameter<HeatFlow> heatFlow = new Parameter<>(unitSystem.getHeatFlowUnit());
+        InletStreamBond<HeatStream>  inletStreamBond = new InletStreamBond<>(idGenerator.getUniqueId());
+        OutletStreamBond<HeatStream> outletStreamBond = new OutletStreamBond<>(idGenerator.getUniqueId());
+        heatStream = new HeatStream(uniqueId, idGenerator, heatFlow, inletStreamBond, outletStreamBond);
+        heatStream.setName("HeatStream " + uniqueId);
+
+        heatStream.setExternalSolver(allSolvers.getHeatStreamSolver());
+        allElements.addHeatStream(heatStream);
+        return heatStream;
+    }
+
+    public FluidStream createDefaultFluidStream(){
+        FluidStream fluidStream;
+        int uniqueId = idGenerator.getUniqueId();
+        InletStreamBond<FluidStream>  inletStreamBond = new InletStreamBond<>(idGenerator.getUniqueId());
+        OutletStreamBond<FluidStream> outletStreamBond = new OutletStreamBond<>(idGenerator.getUniqueId());
+        Fluid specificParameters = new Fluid(allSolvers.getFluidSolver());
+        fluidStream = new FluidStream(uniqueId, idGenerator, specificParameters, inletStreamBond, outletStreamBond);
+        fluidStream.setName("FluidStream " + uniqueId);
+
+        fluidStream.setExternalSolver(allSolvers.getMatterStreamSolver());
+        allElements.addMatterStream(fluidStream);
+        return fluidStream;
+    }
+
+    private Channel createDefaultChannel(Item ownerItem, Parameter<PressureDifference> pressureDrop, ParameterWithDirection heatFlow, ParameterWithDirection extraMassFlow){
+        Channel channel;
+        InletDeviceBond inletDeviceBond = new InletDeviceBond(idGenerator.getUniqueId());
+        OutletDeviceBond outletDeviceBond = new OutletDeviceBond(idGenerator.getUniqueId());
+        channel = new Channel(ownerItem, inletDeviceBond, outletDeviceBond, pressureDrop, heatFlow, extraMassFlow);
+        return channel;
+    }
+
+    public Heater createDefaultHeater(){
+        Heater heater;
+        int uniqueId = idGenerator.getUniqueId();
+        Parameter<PressureDifference> pressureDrop = new Parameter<>(unitSystem.getPressureDifferenceUnit(), 0.0);
+        ParameterWithDirection heatFlow = new ParameterWithDirection(new Parameter(unitSystem.getHeatFlowUnit()), INLET);
+        ParameterWithDirection extraMassFlow = new ParameterWithDirection(new Parameter(unitSystem.getMassFlowUnit(), 0.0), INLET);
+        InletDeviceBond<HeatStream> heatStreamInletDeviceBond= new InletDeviceBond<>(idGenerator.getUniqueId());
+        heater = new Heater(uniqueId, idGenerator, null, heatStreamInletDeviceBond);
```

Comment:
null is bad. don't use it, create another constructor without that field, and on the interface/API side use Optional

https://codecouple.pl/2016/03/09/pozbadz-sie-null-pointerow-java-util-optional/